F841 - False positive

Found this issue while playing with ruff, the variable `exponential` is used

```py
def f():
    exponential, base_multiplier = 1, 2
    hash_map = {
        (exponential := (exponential * base_multiplier) % 3): i + 1 for i in range(2)
    }
    return hash_map

# ruff --select=F test.py
# test.py:2:5: F841 Local variable `exponential` is assigned to but never used
```

--- a/resources/test/fixtures/pyflakes/F841_0.py
+++ b/resources/test/fixtures/pyflakes/F841_0.py
@@ -86,3 +86,11 @@ def f():
         open("") as ((this, that)),
     ):
         print("hello")
+
+
+def f():
+    exponential, base_multiplier = 1, 2
+    hash_map = {
+        (exponential := (exponential * base_multiplier) % 3): i + 1 for i in range(2)
+    }
+    return hash_map

--- a/src/ast/visitor.rs
+++ b/src/ast/visitor.rs
@@ -266,8 +266,8 @@ pub fn walk_expr<'a, V: Visitor<'a> + ?Sized>(visitor: &mut V, expr: &'a Expr) {
             }
         }
         ExprKind::NamedExpr { target, value } => {
-            visitor.visit_expr(target);
             visitor.visit_expr(value);
+            visitor.visit_expr(target);
         }
         ExprKind::BinOp { left, op, right } => {
             visitor.visit_expr(left);

--- a/src/checkers/ast.rs
+++ b/src/checkers/ast.rs
@@ -4449,7 +4449,7 @@ impl<'a> Checker<'a> {
 
     fn check_deferred_assignments(&mut self) {
         self.deferred_assignments.reverse();
-        while let Some((scopes, _parents)) = self.deferred_assignments.pop() {
+        while let Some((scopes, ..)) = self.deferred_assignments.pop() {
             let scope_index = scopes[scopes.len() - 1];
             let parent_scope_index = scopes[scopes.len() - 2];
             if self.settings.rules.enabled(&Rule::UnusedVariable) {

